packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -82,54 +86,120 @@ import {
   QueryBuilderNavigationBlocker,
   type QueryBuilderState,
 } from '@finos/legend-query-builder';
+import { LegendQueryTelemetryHelper } from '../application/LegendQueryTelemetryHelper.js';
+import type { HelpMenuItemEditorRender } from '../stores/LegendQueryApplicationPlugin.js';
 
 const QueryExportDialogContent = observer(
   (props: { exportState: QueryExportState }) => {
     const { exportState } = props;
     const applicationStore = useApplicationStore();
-    const allowCreate = exportState.allowPersist;
+
+    const [isExistingQueryName, setIsExistingQueryName] = useState(false);
+
+    const saveNewQuery = exportState.allowPersist;
     const allowSave = exportState.allowPersist && exportState.allowUpdate;
     const create = applicationStore.guardUnhandledError(() =>
       exportState.persistQuery(true),
     );
-    const save = applicationStore.guardUnhandledError(() =>
+    const updateQuery = applicationStore.guardUnhandledError(() =>
       exportState.persistQuery(false),
     );
+    const renameExport = Boolean(exportState.renameExport);
 
     // name
     const nameInputRef = useRef<HTMLInputElement>(null);
-    const changeName: React.ChangeEventHandler<HTMLInputElement> = (event) =>
+
+    const debouncedLoadQueries = useMemo(
+      () =>
+        debounce((input: string): void => {
+          flowResult(
+            exportState.editorStore.queryLoaderState.loadQueries(input),
```

Comment:
this is bad, we're manipulating the query loader state which is indepdent from this action, we should separate the states

also, the query here is suboptimal, we should search for query with the same name and set the search limit to 1, since we just need to check for duplciated names